src/main/java/org/opensearch/knn/common/KNNConstants.java

```diff
@@ -89,6 +89,8 @@ public class KNNConstants {
     public static final String FAISS_IVF_DESCRIPTION = "IVF";
     public static final String FAISS_FLAT_DESCRIPTION = "Flat";
     public static final String FAISS_PQ_DESCRIPTION = "PQ";
+    public static final String ENCODER_SQFP16 = "SQfp16";
```

Comment:
I think the interface should have the name="sq" and then the num bits as a parameter in the mapping. Did we settle on this somewhere?

Reply:
No thats not what I meant. Currently, the interface for pq looks like:
```
    "encoder":{
      "name":"pq",
      "parameters":{
        "code_size": 8,
        "m": 8
      }
    }
```

however, sq, in this change, it looks like:
```
    "encoder":{
      "name":"SQfp16"
    }
```

This looks inconsistent.

Reply:
I think the encoder name would be `sq` and then we would have a parameter like `code_size=16` or `q_type=fp16`.

Reply:
I see..got your point.

Reply:
@navneet1v @heemin32 What are your thoughts on this?

Reply:
it's better to have the parameters map for the sake of consistency, but I am not aligned with the name suggested here which are `code_size=16` or `q_type=fp16`. what does `q` stands here?

@naveentatikonda can you check how Lucene has exposed the scaler quantization technique?

Reply:
I think Jack might be referring to `q` as `quantization`. 

Lucene is exposing it through codec format, [Lucene99HnswScalarQuantizedVectorsFormat](https://github.com/apache/lucene/blob/main/lucene/core/src/java/org/apache/lucene/codecs/lucene99/Lucene99HnswScalarQuantizedVectorsFormat.java)

Reply:
we have already defined `q` when we put name of the encoder as `sq`. So we can drop `q` there.

Reply:
Yes I was thinking q_type would be the quantized type. But, @navneet1v makes a good point. I think I agree with param as "type" or "dtype" or "data_type" and mapping it to fp16. Currently, in our mapping for lucene 8-bit, we use "data_type": https://opensearch.org/docs/latest/field-types/supported-field-types/knn-vector/#lucene-byte-vector. I think it probably makes sense to be consistent with this and call it "data_type"

Reply:
But, @jmazanec15 in case of lucene 8-bit `data_type` refers to the type of data users are providing. For lucene-byte-vector users will provide byte sized values. But, here we are taking fp32 values as input which might be inconsistent. So, shall we use it as `type` instead of `data_type` here ?

Reply:
I can think of these three options in setting structure.

```
    "encoder":{
      "name":["sqfp16", "sqip8", "sqbp1"]
    }
```

```
    "encoder":{
      "name":"sq",
      "parameters":{
        "sq_type": ["fp16", "ip8", "bp1"]
      }
    }
```

```
    "encoder":{
      "name":"sq",
      "parameters":{
        "sq_type": ["fp", "ip", "bp"]
        "sq_bits": [16, 8, 1]
      }
    }
```

For SQ, the variation is and will be limited to certain values unlike PQ: float 16bits, int 8 bits, binary 1 bit. Also, each type does not share same implementation. In that sense, option 1 make sense. However, if we consider the case where we combines two quantizer `sqpq`, option 1 might not ideal to support it. Therefore, I prefer option 2. 

If we think about combining two quantizer, it is also needed to have a separate parameter name for each quantizer so that we can provide parameter for each quantizer using existing fields. Maybe, we should add one more layer like? 
```
    "encoder":{
      "name":"sq",
      "parameters":{
        "sq": {
          type: ["fp16", "ip8", "bp1"]
        }
      }
    }
```